src/psyclone/domain/lfric/kern_call_acc_arg_list.py

```diff
@@ -213,6 +245,31 @@ def fs_compulsory_field(self, function_space, var_accesses=None):
         # needs the whole field, so we cannot call the base class.
         self.append(function_space.map_name, var_accesses)
 
+    def fs_intergrid(self, function_space, var_accesses=None):
+        '''Add arrays that need to be uploaded for inter-grid kernels.
+        These arrays contain the mapping between fine and coarse meshes.
+
+        :param function_space: the function space associated with the mesh.
+        :type function_space: :py:class:`psyclone.domain.lfric.FunctionSpace`
+        :param var_accesses: optional VariablesAccessInfo instance to store
+            the information about variable accesses.
+        :type var_accesses: Optional[
+            :py:class:`psyclone.core.VariablesAccessInfo`]
+
+        '''
+        # Is this FS associated with the coarse or fine mesh? (All fields
+        # on a given mesh must be on the same FS.)
+        arg = self._kern.arguments.get_arg_on_space(function_space)
+        if arg.mesh == "gh_fine":
+            # For the fine mesh, we need the *whole* dofmap
+            map_name = function_space.map_name
+            self.append(map_name, var_accesses)
+        else:
+            # For the coarse mesh we only need undf and the dofmap for
+            # the current column
+            self.fs_compulsory_field(function_space,
+                                     var_accesses=var_accesses)
+
```

Comment:
With the addition of the news methods, is the implementation of this class now complete?

Reply:
I think so.